crates/vm-superio-ser/src/rtc_pl031.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+//! Provides a wrapper over an `RtcState` that has serialization capabilities.
+//!
+//! This module defines the `RtcStateSer` abstraction which mirrors the
+//! `RtcState` from the base crate, and adds on top of it derives for
+//! the `Serialize`, `Deserialize` and `Versionize` traits.
+
+use serde::{Deserialize, Serialize};
+use versionize::{VersionMap, Versionize, VersionizeResult};
+use versionize_derive::Versionize;
+use vm_superio::RtcState;
+
+/// Wrapper over an `RtcState` that has serialization capabilities.
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize, Versionize)]
+pub struct RtcStateSer {
+    /// The load register.
+    pub lr: u32,
+    /// The offset applied to the counter to get the RTC value.
+    pub offset: i64,
+    /// The MR register.
+    pub mr: u32,
+    /// The interrupt mask.
+    pub imsc: u32,
+    /// The raw interrupt value.
+    pub ris: u32,
+}
+
+impl From<&RtcStateSer> for RtcState {
+    fn from(state: &RtcStateSer) -> Self {
+        RtcState {
+            lr: state.lr,
+            offset: state.offset,
+            mr: state.mr,
+            imsc: state.imsc,
+            ris: state.ris,
+        }
+    }
+}
+
+impl RtcStateSer {
```

Comment:
Is there a reason for implementing `new` here, and from for the reversed operation?

Reply:
Hmm, not really, I think I was under the impression I can't implement `From` here. I replaced with `From`.